lib/features/on_boarding/pages/onboarding_phone_page.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:design_system/design_system.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:grabber/config/routes/routes.dart';
+import 'package:grabber/core/injection.dart';
+import 'package:grabber/core/phone_region.dart';
+import 'package:grabber/features/on_boarding/pages/base_auth/bloc/signup_cubit.dart';
+import 'package:grabber/features/on_boarding/pages/widgets/on_boarding_base_page.dart';
+import 'package:grabber/generated/l10n.dart';
+import 'package:mask_text_input_formatter/mask_text_input_formatter.dart';
+import 'package:phone_number/phone_number.dart';
+import 'package:styled_text/styled_text.dart';
+
+class OnBoardingPhonePage extends StatefulWidget {
+  const OnBoardingPhonePage({super.key});
+
+  @override
+  State<OnBoardingPhonePage> createState() => _OnBoardingPhoneState();
+}
+
+class _OnBoardingPhoneState extends State<OnBoardingPhonePage> {
+  late final TextEditingController _controller;
+  final MaskTextInputFormatter maskFormatter = MaskTextInputFormatter(
+    mask: '(##) #####-####',
+  );
+  final SignupCubit _signupCubit = getIt.get();
+  bool canContinue = false;
+  final FirebaseAuth _auth = FirebaseAuth.instance;
+  final int max = 15;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = TextEditingController();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _controller.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocConsumer<SignupCubit, SignupState>(
+      bloc: _signupCubit,
+      listener: (context, state) {
+        if (state.phoneIsValid) {
+          _auth.createUserWithEmailAndPassword(
+            email: state.email,
+            password: state.password,
+          );
```

Comment:
this should be called at the event of saveUser on cubit.

Also on SignUpCubit add a bool for `userIsCreated` e coloca um listener pra isso